File: features/master/src/main/java/com/nivasafinance/features/master/codemaster/exception/CodeMasterExceptionFactory.java
package com.nivasafinance.features.master.codemaster.exception;

import org.springframework.context.MessageSource;

public class CodeMasterExceptionFactory {
    
    public CodeMasterExceptionFactory(MessageSource messageSource) {
    }
    
    public CodeMasterNotFoundException notFound(String codeName, MessageSource messageSource) {
        return new CodeMasterNotFoundException(codeName, messageSource);
    }
    
    public CodeValueKeyNotFoundException codeValueKeyNotFound(String key, MessageSource messageSource) {
        return new CodeValueKeyNotFoundException(key, messageSource);
    }
    
    public CodeMasterNotFoundException notFoundById(java.util.UUID id, MessageSource messageSource) {
        return new CodeMasterNotFoundException("ID: " + id, messageSource);
    }
    
    public CodeMasterOperationException createFailed(MessageSource messageSource) {
        return new CodeMasterOperationException("error.codemaster.operation.create.failed", messageSource);
    }
    
    public CodeMasterOperationException retrieveEntityFailed(MessageSource messageSource) {
        return new CodeMasterOperationException("error.codemaster.operation.retrieve.failed", messageSource);
    }
    
    public CodeMasterOperationException updateFailed(MessageSource messageSource) {
        return new CodeMasterOperationException("error.codemaster.operation.update.failed", messageSource);
    }
    
    public CodeMasterOperationException deleteFailed(MessageSource messageSource) {
        return new CodeMasterOperationException("error.codemaster.operation.delete.failed", messageSource);
    }
}

File: features/master/src/main/java/com/nivasafinance/features/master/codemaster/exception/CodeMasterExceptionFactory.kt
package com.nivasafinance.features.master.codemaster.exception

import org.springframework.context.MessageSource

object CodeMasterExceptionFactory {

    fun notFound(codeName: String, messageSource: MessageSource): CodeMasterNotFoundException {
        return CodeMasterNotFoundException(codeName, messageSource)
    }

    fun codeValueKeyNotFound(key: String, messageSource: MessageSource): CodeValueKeyNotFoundException {
        return CodeValueKeyNotFoundException(key, messageSource)
    }

    fun notFoundById(id: java.util.UUID, messageSource: MessageSource): CodeMasterNotFoundException {
        return CodeMasterNotFoundException("ID: $id", messageSource)
    }

    fun createFailed(messageSource: MessageSource): CodeMasterOperationException {
        return CodeMasterOperationException("error.codemaster.operation.create.failed", messageSource)
    }

    fun retrieveEntityFailed(messageSource: MessageSource): CodeMasterOperationException {
        return CodeMasterOperationException("error.codemaster.operation.retrieve.failed", messageSource)
    }

    fun updateFailed(messageSource: MessageSource): CodeMasterOperationException {
        return CodeMasterOperationException("error.codemaster.operation.update.failed", messageSource)
    }

    fun deleteFailed(messageSource: MessageSource): CodeMasterOperationException {
        return CodeMasterOperationException("error.codemaster.operation.delete.failed", messageSource)
    }
}
